util/progress/progressui/display.go

```diff
@@ -69,53 +88,208 @@ func DisplaySolveStatus(ctx context.Context, c console.Console, w io.Writer, ch
 		}
 	}
 
-	var done bool
-	ticker := time.NewTicker(tickerTimeout)
-	// implemented as closure because "ticker" can change
-	defer func() {
-		ticker.Stop()
-	}()
+	d.disp.init(displayTimeout)
+	defer d.disp.done()
 
-	displayLimiter := rate.NewLimiter(rate.Every(displayTimeout), 1)
+	ticker := time.NewTicker(tickerTimeout)
+	defer ticker.Stop()
 
-	var height int
-	width, _ := disp.getSize()
+	var warnings []client.VertexWarning
 	for {
 		select {
 		case <-ctx.Done():
 			return nil, ctx.Err()
 		case <-ticker.C:
+			d.disp.refresh()
 		case ss, ok := <-ch:
-			if ok {
-				t.update(ss, width)
-			} else {
-				done = true
+			if !ok {
+				return warnings, nil
 			}
-		}
 
-		if modeConsole {
-			width, height = disp.getSize()
-			if done {
-				disp.print(t.displayInfo(), width, height, true)
-				t.printErrorLogs(c)
-				return t.warnings(), nil
-			} else if displayLimiter.Allow() {
-				ticker.Stop()
-				ticker = time.NewTicker(tickerTimeout)
-				disp.print(t.displayInfo(), width, height, false)
-			}
-		} else {
-			if done || displayLimiter.Allow() {
-				printer.print(t)
-				if done {
-					t.printErrorLogs(w)
-					return t.warnings(), nil
-				}
-				ticker.Stop()
-				ticker = time.NewTicker(tickerTimeout)
+			d.disp.update(ss)
+			for _, w := range ss.Warnings {
+				warnings = append(warnings, *w)
 			}
+			ticker.Reset(tickerTimeout)
+		}
+	}
+}
+
+func (d SolveStatusDisplay) Warnings() []client.VertexWarning {
+	return nil
```

Comment:
🤔 is this some leftover?

Reply:
Yea this is a leftover. I was originally going to consolidate the warnings into the struct, but it didn't work and I went back to returning them from `UpdateFrom`. This just didn't get removed.